### PR TITLE
feat(ui): Allow setting `resizeToAvoidBottomInset` from LoginScreen and set as default `false` for backwards compatibility.

### LINKED ISSUE #8084: [FlutterFire_UI]: SignInScreen blocked by keyboard on small sized mobile devices
Simple to reproduce this:
1) Create a SigninScreen widget with an image that takes up only the top 25% of the screen
2) Attempt to use it on any medium or small sized Android/iOS device.
3) The password field will be blocked by the keyboard.  You can still type a password, but won't be able to see your typing.  

Please make the contents of this screen scrollable so that the user can scroll within the remaining screen space above the keyboard.

### Discussed in https://github.com/FirebaseExtended/flutterfire/discussions/8050

<div type='discussions-op-text'>

<sup>Originally posted by **fredecodes** February  9, 2022</sup>
There seems to be a really big problem with SigninScreen that I've not seen mentioned elsewhere, which surprises me as I see it as a huge usability issue.  When rendered on a regular or small sized mobile screen, the "Password" text field is completely hidden by the keyboard (on both Android and iOS).  You can still type to enter text, but you are typing in the blind because you can't see the cursor.  This is very frustrating for users, especially when it's their very first impression of your app.  When the keyboard is up, the screen should scroll to fit within the upper bounds that are visible.  Am I missing something, is there a way around this to let the user see the text field that is hidden by the keyboard?</div>

### PR DESCRIPTION
**Pull request by ssabdb**
## Description

The UniversalScaffold used by LoginScreen() and SignInScreen() (among others) sets resizeToAvoidBottomInset=false, which means on smaller screen devices (particularly on mobile web, which pulls in relatively high resolution devices eg. the Pixel 4a) and means the password box is obscured when a header is set.

This change is to expose resizeToAvoidBottomInset as an option which can be set directly from the externally exposed api points, while not breaking any of the expected behaviour elsewhere unless explicit.

There are other screens I could implement this on as well (eg. sign in link.) 

Happy to provide screenshots if necessary.

## Related Issues

Closes #8084

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/flutterfire_ui/lib/src/auth/screens/sign_in_screen.dart [hunk 1]
```diff
@@ -80,12 +82,14 @@ class SignInScreen extends MultiProviderScreen {
...
         providerConfigs: providerConfigs,
         auth: auth,
+        resizeToAvoidBottomInset: resizeToAvoidBottomInset,
         headerMaxExtent: headerMaxExtent,
         headerBuilder: headerBuilder,
         sideBuilder: sideBuilder,
         desktopLayoutDirection: desktopLayoutDirection,
         oauthButtonVariant: oauthButtonVariant,
         email: email,
+        resizeToAvoidBottomInset: resizeToAvoidBottomInset,
```
Comment (russellwheatley) [inline packages/flutterfire_ui/lib/src/auth/screens/sign_in_screen.dart:92]: Is this one time too many?

**Review by russellwheatley — DISMISSED** [review]
Seems fair to me albeit with one question.

**Review by lesnitsky — APPROVED** [review]

File: packages/flutterfire_ui/lib/src/auth/screens/sign_in_screen.dart [hunk 2]
```diff
@@ -80,12 +82,14 @@ class SignInScreen extends MultiProviderScreen {
         action: AuthAction.signIn,
         providerConfigs: providerConfigs,
         auth: auth,
+        resizeToAvoidBottomInset: resizeToAvoidBottomInset,
```
Comment (Salakar) [inline packages/flutterfire_ui/lib/src/auth/screens/sign_in_screen.dart:85]:
```suggestion
```

**Comment by ssabdb** [comment]
Thanks for finishing it @Salakar.

**Pull request merged**